eng/pipelines/common/maui-templates.yml

```diff
@@ -6,96 +6,158 @@ parameters:
-              inputs:
-                artifactName: ${{ parameters.artifactName }}
-                itemPattern: ${{ parameters.artefactItemPattern }}
-                downloadPath: $(System.DefaultWorkingDirectory)/artifacts
-
-            - pwsh: Move-Item -Path artifacts\nuget\*.nupkg -Destination artifacts -Force
-              displayName: Move the downloaded artifacts
-
-            - pwsh: ./build.ps1 --target=dotnet-local-workloads --configuration="${{ BuildConfiguration }}" --verbosity=diagnostic
-              displayName: 'Install .NET (Local Workloads)'
-              retryCountOnTaskFailure: 3
-              env:
-                DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
-                PRIVATE_BUILD: $(PrivateBuild)
-
-            - pwsh: ./build.ps1 --target=dotnet-templates --configuration="${{ BuildConfiguration }}" --testtfm="${{ TestTFM.tfm }}" --verbosity=diagnostic
-              displayName: 'Build .NET MAUI Templates'
-
-            - task: PublishBuildArtifacts@1
-              condition: always()
-              displayName: publish artifacts
-              inputs:
-                ArtifactName: '${{ parameters.uploadPrefix }}${{ BuildPlatform.artifact }}'
+- ${{ each BuildPlatform in parameters.BuildPlatforms }}:
+  - job: build_${{ BuildPlatform.name }}
+    workspace:
+      clean: all
+    displayName: ${{ BuildPlatform.name }}
+    timeoutInMinutes: 240
+    condition: ${{ parameters.condition}}
+    pool:
+      name: ${{ BuildPlatform.poolName }}
+      vmImage: ${{ BuildPlatform.vmImage }}
+      demands:
+        - macOS.Name -equals Monterey
+        - macOS.Architecture -equals x64
+        - Agent.HasDevices -equals False
+        - Agent.IsPaired -equals False
+    steps:
+    - template: provision.yml
+      parameters:
+        platform: ${{ BuildPlatform.name }}
+
+    - task: DownloadBuildArtifacts@0
+      displayName: 'Download Packages'
+      inputs:
+        artifactName: ${{ parameters.artifactName }}
+        itemPattern: ${{ parameters.artifactItemPattern }}
+        downloadPath: $(System.DefaultWorkingDirectory)/artifacts
+
+    - pwsh: Move-Item -Path artifacts\nuget\*.nupkg -Destination artifacts -Force
+      displayName: Move the downloaded artifacts
+
+    - pwsh: ./build.ps1 --target=dotnet-local-workloads --verbosity=diagnostic
+      displayName: 'Install .NET (Local Workloads)'
+      retryCountOnTaskFailure: 3
+      env:
+        DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
+        PRIVATE_BUILD: $(PrivateBuild)
+
+    - task: DotNetCoreCLI@2
+      inputs:
+        projects: $(Build.SourcesDirectory)/tests/Microsoft.Maui.IntegrationTests/Microsoft.Maui.IntegrationTests.csproj
+      displayName: Build Microsoft.Maui.IntegrationTests
+
+    - task: DotNetCoreCLI@2
+      inputs:
+        command: test
+        projects: $(Build.SourcesDirectory)/tests/Microsoft.Maui.IntegrationTests/bin/Debug/net7.0/Microsoft.Maui.IntegrationTests.dll
```

Comment:
is this always net7.0?

Reply:
Yeah this is based on the TF of the new project which can stay on a stable/LTS version of .NET.